app/src/main/java/com/hamdan/forzenbook/data/repository/LoginRepositoryImpl.kt

```diff
@@ -7,13 +7,16 @@ class LoginRepositoryImpl(
     private val dao: LoginDao,
     private val service: LoginService
 ) : LoginRepository {
-    override suspend fun getToken(email: String): LoginData? {
-        //check database for token
-        //if no token in data base get from service
+    override suspend fun requestValidation(email: String): Boolean {
         return try {
-            LoginData(service.getToken().body()?.token)
+            service.requestValidation(email).isSuccessful
         } catch (e: Exception) {
-            null
+            throw (e)
```

Comment:
Why catch just to rethrow with nothing else?

Reply:
I was thinking that adding the structure ahead of time would be good incase we want to do something here, however with how it is currently there is no need for a catch at this level if the usecase will catch anyway.